src/task.rs

```diff
@@ -463,7 +463,7 @@ pub mod watchdog {
             TWDT_DRIVER_REF_COUNT.fetch_add(1, Ordering::SeqCst);
             let init_by_idf = Self::watchdog_is_init_by_idf();
 
-            #[cfg(not(esp_idf_version_major = "4"))]
+            #[cfg(all(not(esp_idf_version_major = "4"), not(esp_idf_version = "5.0")))]
```

Comment:
As I checked the docs, `esp_task_wdt_reconfigure()` is in [release/5.0](https://docs.espressif.com/projects/esp-idf/en/release-v5.0/esp32/api-reference/system/wdts.html#_CPPv424esp_task_wdt_reconfigurePK21esp_task_wdt_config_t). And I think the CI of this crate covers the 5.0 build and runs fine. How do you specify the idf version in your `.cargo/Config`?

Reply:
Interesting you are right the SDK documentation stating it is in release/5.0, but not in stable (v5.0) - whatever the difference is ... 

I've specified the idf version in following way:
```
[env]
ESP_IDF_VERSION = { value = "branch:release/v5.0" }
```
When using ```branch:release/v5.0``` than ```esp_task_wdt_reconfigure()``` is not found.

Reply:
Very strange. In the repo on branch `release/v5.0` [`esp_task_wdt_reconfigure()`](https://github.com/espressif/esp-idf/blob/490216a2ace6dc3e1b9a3f50d265a80481b32f6d/components/esp_system/include/esp_task_wdt.h#L63) exists, and I guess with `ESP_IDF_VERSION = { value = "branch:release/v5.0" }` this branch should be pulled.

@ivmarkov do you have an explanation for this.

Reply:
@taunusflieger The reason is likely that you had an older 5.0 branch in your `<binary crate>/.espressif` directory. Usually, deleting this directory helps (or pulling from the branch manually).

(BTW: the ESP IDF installed by `espup` has nothing to do with the ESP IDF branch used by `esp-idf-sys` and IMO brings more confusion that actually solving any problem.)